PKHeX/Legality/Checks.cs

```diff
@@ -2503,7 +2503,7 @@ private CheckResult[] parseMovesWasEggPreRelearn(int[] Moves, List<int>[] validL
                 for (int i = 0; i <= splitctr; i++)
                 {
                     var LvlupEggMoves = Legal.getBaseEggMoves(pkm, i, ver, 100)?.ToArray() ?? new int[0];
-                    var EggMoves = !pkm.WasEventEgg && !pkm.WasGiftEgg ? Legal.getEggMoves(pkm, i, ver).ToArray() : new int[0];
+                    var EggMoves = Legal.getEggMoves(pkm, i, ver).ToArray();
```

Comment:
Note here egg moves should be intentionally put into `parseMoves` for incompatible checking.